### PR TITLE
First lemma in lcm inequality proof

### PR DESCRIPTION
**Pull request by metakunt**
I've figured out how to do https://github.com/metamath/set.mm/issues/3947

### CODE REVIEW

**Comment by metakunt** [comment]
As promised, I won't push any more changes unless they are requested.

File: set.mm
Comment (tirix) [inline set.mm:632540]: I believe the first hypothesis:
```
|- ( ( ph /\ x e. A ) -> F e. CC )
```
Can be derived from the second:
```
|- ( ph -> ( x e. A |-> F ) e. ( A -cn-> CC ) )
```
See for example ~ cncff

Comment (metakunt) [reply set.mm:632540]: Ah, well time to fix it.

**Review by tirix — APPROVED** [review]
Wow, that's a lot of progress since last week! Way to go!

**Comment by metakunt** [comment]
Thanks a lot. Yeah metamath has some insane cliffs you have to jump over. Once you understand which theorems help and learn the diverse equality inference theorems it gets easier.

Honestly, I'll keep the hypotheses on the sublemmas and only remove it on the final lemma since the sublemmas shouldn't be used anyways.

File: set.mm
Comment (tirix) [reply set.mm:632540]: I would have fixed them from the last to the first, so that not everything is broken.

For fixing a theorem, just reopen the proof file, remove the "first" hypothesis, and fill the blank.
This should work:
```
h10::         |- ( ph -> ( x e. A |-> F ) e. ( A -cn-> CC ) )
20::cncff            |- ( ( x e. A |-> F ) e. ( A -cn-> CC ) -> ( x e. A |-> F ) : A --> CC )
30:10,20:syl        |- ( ph -> ( x e. A |-> F ) : A --> CC )
40:30:fvmptelrn   |- ( ( ph /\ x e. A ) -> F e. CC )
```

**Comment by metakunt** [comment]
Ah cool, it's fixed. I did it manually by deleting the hypotheses in set.mm and rewriting the proof. It's fine that the hypotheses stay in 1-4, those lemmas shouldn't be used anyways. Now I can prove the equation
```math
\int_0^1 x^{m-1} \sum_{k=0}^{n-m} (-1)^{k} \binom{n-m}{k}\cdot x^k =
 \sum_{k=0}^{n-m} (-1)^{k} \binom{n-m}{k} \int_0^1 x^{m-1}\cdot x^k
```

Now I need to satisfy the hypotheses 1-7:
A = [0,1] Ok
B = ( 0 ... ( N - M ) ) e. Fin Ok
L = 0 -> 0 e. RR Ok
U = 1 -> 1 e. RR Ok
The difficulty will be satisfying the final 3.
I can use the binomial coefficient operation to get  https://us.metamath.org/mpeuni/df-bc.html
I guess G = ( ( ( -u 1 ) ^ k ) x. ( ( N - M ) _C k ) )
so that one should be fine. The problem will be F and H. I will need to prove that both are continuous on the restriction where I've defined it.

Also later on I will need several helpful lemmas regarding differentiation and integration. I know that those already exist, but I still would like to know what the easiest way would be to get started quickly. I need continuous function builders and ways to integrate easily, what are the lemmas I should focus on learning now?

**Comment by tirix** [comment]
> The problem will be F and H. I will need to prove that both are continuous

So you have $F(x) = x^{m-1}$ and $H(x) = x^k$, right?
I believe [`~cxpcncf1`](https://us.metamath.org/mpeuni/cxpcncf1.html) should work for both.
Or simpler: [`~expcncf`](https://us.metamath.org/mpeuni/expcncf.html).

File: set.mm
Comment (avekens) [inline set.mm:632632]: What is the meaning of this equation? It should be described in more detail (especially since the  "lcm inequality proof" is not available yet).

Comment (metakunt) [reply set.mm:632632]: I have added documentation outlining the idea behind this theorem

Comment (tirix) [reply set.mm:632632]: The link seems to be missing a tilde ~ : [https://www3.nd.edu/~andyp/notes/AKS.pdf](https://www3.nd.edu/~andyp/notes/AKS.pdf)

Comment (metakunt) [reply set.mm:632632]: I added the tilde back, but after rewrapping it just looks wrong. I don't know how metamath.exe handles tildes in links

Comment (metakunt) [reply set.mm:632632]: I might have to change tilde to %7e, don't I?

Comment (wlammen) [reply set.mm:632632]: %7e works, at least

**Review by wlammen — APPROVED** [review]

**Comment by tirix** [comment]
This looks good to me, can we merge once we have @avekens's final approval?

File: set.mm
Comment (avekens) [reply set.mm:632632]: To be honest, I did not find this theorem in the referenced documentation. 
But maybe everything becomes clearer after more theorems are available for this topic. 

Since this is currently in a mathbox only, I will accept PR.

**Review by avekens — APPROVED** [review]

File: set.mm
Comment (tirix) [reply set.mm:632632]: I saw the equations on which @metakunt is working in the proof of theorem 3.1.

**Pull request merged**
